compiler/rustc_infer/src/infer/error_reporting/suggest.rs

```diff
@@ -412,14 +412,70 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
                 let found_sig =
                     &(self.normalize_fn_sig)(self.tcx.bound_fn_sig(*did2).subst(self.tcx, substs2));
 
-                if self.same_type_modulo_infer(*found_sig, *expected_sig) {
-                    diag.note(
+                diag.note(
                     "different fn items have unique types, even if their signatures are the same",
-                    );
+                );
```

Comment:
I think the condition should remain around this note, as otherwise we're emitting the diagnostic when it's not relevant—for instance you now have it appearing in `tests/ui/issues/issue-59488.stderr` when it need not?

Reply:
I agree